pkg/ingress/ingress.go

```diff
@@ -162,6 +172,14 @@ func (i *Ingress) IngressClass() string {
 	return strings.ToLower(val)
 }
 
+func (i *Ingress) IngressProvider() string {
+	val, ok := i.IngressApi.Annotations[kubelego.AnnotationIngressProvider]
+	if !ok {
+		return i.kubelego.LegoDefaultIngressProvider()
```

Comment:
After some more thought, I've realised this may break current implementations of the GCE ingress controller - please do correct me if I'm wrong...

If a user currently has an ingress that specifies
`kubernetes.io/ingress.class: "gce"`, and no `ingress.provider` annotation (as they are upgrading), the `ingress.provider` will default to `nginx`, thus breaking their set up.

To work around this and perform these changes in a backwards compatible way, I think we need to set the `ingress.provider` value to `ingress.class` if it's not already defined on the ingress resource itself.